src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -55,30 +54,56 @@
      * @param type The class type to be registered
      * @deprecated use {@link RegisterCapabilitiesEvent}
      */
-    @Deprecated(forRemoval = true)
+    @Deprecated(since = "1.18", forRemoval = true)
     public <T> void register(Class<T> type)
     {
         Objects.requireNonNull(type,"Attempted to register a capability with invalid type");
-        String realName = type.getName().intern();
+        get(Type.getType(type).toString(), true);
+    }
+
+
+    public static <T> Capability<T> get(CapabilityToken<T> type)
+    {
+        return INSTANCE.get(type.getType(), false);
+    }
+
+    @SuppressWarnings("unchecked")
+    <T> Capability<T> get(String realName, boolean registering)
+    {
         Capability<T> cap;
 
         synchronized (providers)
         {
-            if (providers.containsKey(realName)) {
+            realName = realName.intern();
+            cap = (Capability<T>)providers.get(realName);
+            if (cap == null)
+            {
+                cap = new Capability<>(realName);
+                providers.put(realName, cap);
+            }
+
+            if (cap.isRegistered() && registering)
```

Comment:
This call to `isRegistered` can technically fail, because it's not guarded by a `synchronized` on the `Capability`.